src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -369,8 +375,12 @@ public static void checkApiUrlValidity(@Nonnull GitHub gitHub, @CheckForNull Sta
 
                 GitHubBuilder gb = createGitHubBuilder(apiUrl, cache);
 
-                if (username != null) {
-                    gb.withPassword(username, password);
+                if (username != null && password != null) {
+                    gb.withAuthorizationProvider(ImmutableAuthorizationProvider.fromOauthToken(password, username));
+                }
+
+                if (gitHubAppCredentials != null) {
+                    gb.withAuthorizationProvider(gitHubAppCredentials.getAuthorizationProvider());
```

Comment:
```suggestion
                if (username != null) {
                    gb.withPassword(username, password);
                } 

                // For github app, use the auto-refresh provider
                if (credentials instanceof GitHubAppCredentials) {
                    gb.withAuthorizationProvider(((GitHubAppCredentials) credentials).getAuthorizationProvider());
```

Reply:
without adding the `&& password != null` this will call withPassword when and authorization provider is set for github app, not sure if we want that

Reply:
That is what we currently do and it works.